micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemoryMetrics.java

```diff
@@ -100,6 +98,58 @@ public void bindTo(MeterRegistry registry) {
                 .baseUnit(BaseUnits.BYTES)
                 .register(registry);
         }
+
+        //The used and committed size of the returned memory usage is the sum of those values of all heap/non-heap memory pools
+        // whereas the max size of the returned memory usage represents the setting of the non-heap/non-heap memory which may
+        // not be the sum of those of all non-heap memory pools. If the setting is missing for heap/non-heap memory pools these values come as -1.
+
+        MemoryMXBean memoryMXBean = ManagementFactory.getMemoryMXBean();
+
+        Iterable<Tag> tagsWithId = Tags.concat(tags,"id", "total", "area", "heap");
+
+        Gauge.builder("jvm.memory.used", memoryMXBean, (mem) -> getTotalAreaUsageValue(mem, MemoryUsage::getUsed, "heap"))
+                .tags(tagsWithId)
+                .description("The amount of used memory")
+                .baseUnit(BaseUnits.BYTES)
+                .register(registry);
+
+        Gauge.builder("jvm.memory.max", memoryMXBean, (mem) -> getTotalAreaUsageValue(mem, MemoryUsage::getMax, "heap"))
+                .tags(tagsWithId)
+                .description("The maximum amount of memory in bytes that can be used for memory management")
+                .baseUnit(BaseUnits.BYTES)
+                .register(registry);
+
+        Gauge.builder("jvm.memory.committed", memoryMXBean, (mem) -> getTotalAreaUsageValue(mem, MemoryUsage::getCommitted, "heap"))
+                .tags(tagsWithId)
+                .description("The amount of memory in bytes that is committed for the Java virtual machine to use")
+                .baseUnit(BaseUnits.BYTES)
+                .register(registry);
+
+        Gauge.builder("jvm.memory.heap_used_percent", memoryMXBean, (mem) -> getHeapUsagePercent(mem))
```

Comment:
is there a reason we need to precompute the percent when we are shipping its constituent parts?

Reply:
We are using Statsd meter registry to send metrics to InfluxDB via Telegraf agent and we use Grafana for visualising these metrics. As cross measurement calculations are not directly supported in InfluxDB, this percentage calculation is not possible when we get the "jvm_memory_used" and "jvm_memory_max" in different measurements. 

Also when we use Kapacitor for alerts on such calculations, we need to perform joins for the the sums of "jvm_memory_used" and "jvm_memory_max" for individual pools. When there are even few missing values for any of the pools in either "jvm_memory_used" or "jvm_memory_max", it is resulting in either wrong alerts being raised or already raised alerts being reset in Kapacitor. 

To avoid all these calculations at different levels, it is better to precompute the percent.

Reply:
I want to try to come up with a solution for this for you. But pre-computing doesn't seem to be the right way, or we'd be adding a ton of this throughout instrumentation everywhere. You're adding pre-computed stats for things you care about now, but there's no logical end to this progression. Can you help me understand why Influx division doesn't work to compute percentiles? https://docs.influxdata.com/influxdb/v1.8/query_language/math_operators/#division

Reply:
i.e. is it because they aren't from the same `FROM xxx`? Is this a limitation of Influx or of Grafana's UI implementation of Influx queries?

Reply:
Yes, it is because they are not from same `FROM xxx`. This is a limitation of Influx, not the Grafana.